File: src/main/java/com/example/appjamback/global/error/exception/ErrorCode.java
package com.example.appjamback.global.error.exception;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum ErrorCode {

    EXPIRED_JWT(401, "Expired Jwt"),
    INVALID_JWT(401, "Invalid Jwt"),
    PASSWORD_MISS_MATCH(401, "Password Miss Match"),
    INVALID_REFRESH_TOKEN(401, "Invalid Refresh Token"),

    ACCOUNTID_ALREADY_EXISTS(409, "AccountId Already Exists"),
    USER_NOT_FOUND(404, "User Not Found"),
    IMAGE_NOT_FOUND(404, "Image Not Found"),

    INTERNAL_SERVER_ERROR(500, "Internal Server Error");

    private final int status;
    private final String message;
}

File: src/main/kotlin/com/example/instargram/global/error/exception/ErrorCode.kt
package com.example.instargram.global.error.exception

enum class ErrorCode (
    val status: Int,
    val message: String
){
    INTERNAL_SERVER_ERROR(500, "Internal Server Error"),

    INVALID_TOKEN(401,"Invalid Token"),
    EXPIRED_TOKEN(401, "Expired Token"),

    ACCOUNTID_NOT_FOUND(404, "AccountId Not Found"),
    ACCOUNTID_ALREADY_EXISTS(409, "AccountId Already Exists"),

    PASSWORD_MISS_MATCH(400, "Password Miss Match"),
    USER_NOT_FOUND(404, "User Not Found"),

}
